ZenPacks/zenoss/HttpMonitor/http.py

```diff
@@ -186,6 +230,15 @@ def _answer(self, body=""):
             regex = self._checkRegex(body)
             if regex and regex.get('status', ""):
                 res['msg'] = regex
+
+        if self._follow in ('ok', 'fail'):
+            if self._follow == "ok":
+                # we don't have to do anything with it
+                pass
+            if self._follow == "fail":
+                if self._response.code in (301,302,303,307,308):
+                    res['msg'] = {'status': 'CRITICAL', 'msg': ''}
```

Comment:
I think this logic draws no real distinction between the "ok" and "follow" directives.  If the directive is "ok", then *any* non-error response is a positive result.  If the directive is "follow", then, again, *any* non-error response is a positive result.

This will require looking at `check_http`'s behavior to be sure, but I assumed that the "ok" directive *requires* a redirect response.  This would distinguish it from "follow", which *does not require* a redirect response.  Without this distinction, I don't see any useful difference between "follow" and "ok".

Reply:
On some sites could be 'endless redirection' and we won't follow redirection and just returns OK.
for example with `OK`:
(zenoss) [zenoss@99db588dfebb ~]$ /usr/lib64/nagios/plugins/check_http -H password.zenoss.loc -f ok
HTTP OK: HTTP/1.1 302 Found - 358 bytes in 0.272 second response time |time=0.272098s;;;0.000000 size=358B;;;0

with option `follow`:
(zenoss) [zenoss@99db588dfebb ~]$ /usr/lib64/nagios/plugins/check_http -H password.zenoss.loc -f follow
HTTP WARNING - maximum redirection depth 15 exceeded - https://password.zenoss.loc:443/home.html

For this cases, we have to keep both options.


If with the option `OK` you catch return code like `500` error going to be raised.